### PR TITLE
Erce/bump dependencies

### PR DESCRIPTION
**Pull request by ercecan**
# Description

## Linked Issues
- Fixes # (issue, if applicable)
- Related to # (issue) 

## Testing
Describe how these changes were tested. If you've added new features, have you added unit tests?

## Docs
Describe where this code is documented. If it changes a documented interface, have the docs been updated?

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1168,10 +1231,22 @@ dependencies = [
 ]
 
 [[package]]
-name = "binout"
-version = "0.2.1"
+name = "bindgen"
```
Comment (kpp) [inline Cargo.lock:1234]: That is questionable

File: Cargo.lock [hunk 2]
```diff
@@ -1954,7 +2029,7 @@ dependencies = [
  "anyhow",
  "backoff",
  "bincode",
- "bonsai-sdk",
+ "bonsai-sdk 0.9.0",
```
Comment (kpp) [inline Cargo.lock:2032]: Why not 1.1.2?

File: Cargo.lock [hunk 3]
```diff
@@ -3780,6 +3933,26 @@ dependencies = [
  "serde",
 ]
 
+[[package]]
+name = "inotify"
```
Comment (kpp) [inline Cargo.lock:3937]: Why?

File: Cargo.lock [hunk 4]
```diff
@@ -4168,18 +4341,23 @@ dependencies = [
...
- "once_cell",
- "serde",
- "serde_derive",
- "serde_yaml",
+ "kqueue-sys",
+ "libc",
+]
+
+[[package]]
+name = "kqueue-sys"
```
Comment (kpp) [inline Cargo.lock:4354]: Why?

File: Cargo.lock [hunk 5]
```diff
@@ -4413,6 +4601,21 @@ dependencies = [
  "libc",
 ]
 
+[[package]]
+name = "metal"
```
Comment (kpp) [inline Cargo.lock:4605]: Why?

File: Cargo.lock [hunk 6]
```diff
@@ -4711,6 +4958,15 @@ dependencies = [
  "smallvec",
 ]
 
+[[package]]
+name = "objc"
```
Comment (kpp) [inline Cargo.lock:4962]: Why?

File: Cargo.lock [hunk 7]
```diff
@@ -5098,20 +5434,47 @@ dependencies = [
...
+ "itertools 0.13.0",
+ "proc-macro2",
+ "quote",
+ "syn 2.0.72",
+]
 
 [[package]]
-name = "quinn"
-version = "0.11.3"
+name = "puffin"
```
Comment (kpp) [inline Cargo.lock:5450]: Why?

File: Cargo.lock [hunk 8]
```diff
@@ -5423,11 +5746,46 @@ dependencies = [
  "winreg 0.52.0",
 ]
 
+[[package]]
+name = "reth-beacon-consensus"
```
Comment (kpp) [inline Cargo.lock:5750]: Why?

File: bin/citrea/provers/risc0/guest-mock/Cargo.lock [hunk 9]
```diff
@@ -1456,12 +1638,36 @@ version = "0.4.21"
...
+]
+
 [[package]]
 name = "memchr"
 version = "2.7.4"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "78ca9ab1a0babb1e7d5695e3530886289c18cf2f87ec19a575a0abdce112e3a3"
 
+[[package]]
+name = "metal"
```
Comment (kpp) [inline bin/citrea/provers/risc0/guest-mock/Cargo.lock:1657]: Suspicious

File: bin/citrea/provers/risc0/guest-mock/Cargo.lock [hunk 10]
```diff
@@ -840,6 +950,33 @@ dependencies = [
  "unicode-segmentation",
 ]
 
+[[package]]
+name = "core-foundation"
```
Comment (kpp) [inline bin/citrea/provers/risc0/guest-mock/Cargo.lock:954]: Suspicious

File: crates/ethereum-rpc/src/gas_price/fee_history.rs [hunk 11]
```diff
@@ -72,7 +73,13 @@ impl<C: sov_modules_api::Context> FeeHistoryCache<C> {
     }
 
     /// Processing of the arriving blocks
-    pub fn insert_blocks(&mut self, blocks: Vec<(Rich<Block>, Vec<AnyTransactionReceipt>)>) {
+    pub fn insert_blocks(
+        &mut self,
+        blocks: Vec<(
+            WithOtherFields<Block<WithOtherFields<Transaction>>>,
```
Comment (kpp) [inline crates/ethereum-rpc/src/gas_price/fee_history.rs:79]: Find some predefined type like https://docs.rs/alloy/latest/alloy/rpc/types/type.AnyTransactionReceipt.html

File: crates/ethereum-rpc/src/gas_price/fee_history.rs [hunk 12]
```diff
@@ -166,7 +173,7 @@ pub(crate) fn calculate_reward_percentiles_for_block(
     percentiles: &[f64],
     gas_used: u64,
     base_fee_per_gas: u64,
-    transactions: &[Transaction],
+    transactions: &[WithOtherFields<Transaction>],
```
Comment (kpp) [inline crates/ethereum-rpc/src/gas_price/fee_history.rs:176]: https://docs.rs/alloy/latest/alloy/network/struct.AnyNetwork.html#associatedtype.TransactionResponse

File: crates/ethereum-rpc/src/subscription.rs [hunk 13]
```diff
@@ -76,7 +77,7 @@ impl Drop for SubscriptionManager {
 }
 
 pub async fn new_heads_notifier(
-    mut rx: mpsc::Receiver<RichBlock>,
+    mut rx: mpsc::Receiver<WithOtherFields<Block<WithOtherFields<Transaction>>>>,
```
Comment (kpp) [inline crates/ethereum-rpc/src/subscription.rs:80]: AnyNetworkBlock

File: crates/ethereum-rpc/src/subscription.rs [hunk 14]
```diff
@@ -133,7 +134,7 @@ pub async fn logs_notifier(
 pub async fn soft_confirmation_event_handler<C: sov_modules_api::Context>(
     storage: C::Storage,
     mut soft_confirmation_rx: broadcast::Receiver<u64>,
-    new_heads_tx: mpsc::Sender<RichBlock>,
+    new_heads_tx: mpsc::Sender<WithOtherFields<Block<WithOtherFields<Transaction>>>>,
```
Comment (kpp) [inline crates/ethereum-rpc/src/subscription.rs:137]: AnyNetworkBlock

File: crates/evm/src/evm/error/pool.rs [hunk 15]
```diff
@@ -215,18 +225,19 @@ impl InvalidPoolTransactionError {
     /// Returns `true` if the error was caused by a transaction that is considered bad in the
     /// context of the transaction pool and warrants peer penalization.
     ///
-    /// See [PoolError::is_bad_transaction].
+    /// See [`PoolError::is_bad_transaction`].
+    #[allow(clippy::match_same_arms)]
```
Comment (kpp) [inline crates/evm/src/evm/error/pool.rs:229]: No

File: crates/evm/src/evm/system_events.rs [hunk 16]
```diff
@@ -1,16 +1,21 @@
...
-use super::system_contracts::{BitcoinLightClient, Bridge};
+use super::system_contracts::{BitcoinLightClient, BridgeWrapper};
 
 /// This is a special signature to force tx.signer to be set to SYSTEM_SIGNER
-pub const SYSTEM_SIGNATURE: Signature = Signature {
-    r: U256::ZERO,
-    s: U256::ZERO,
-    odd_y_parity: false,
-};
+pub static SYSTEM_SIGNATURE: Lazy<Signature> = Lazy::new(|| {
```
Comment (kpp) [inline crates/evm/src/evm/system_events.rs:12]: Why?

File: crates/evm/src/query.rs [hunk 17]
```diff
@@ -606,7 +635,7 @@ impl<C: sov_modules_api::Context> Evm<C> {
                 _ => {
                     let block = self
                         .get_sealed_block_by_number(block_number, working_set)?
-                        .ok_or(EthApiError::UnknownBlockNumber)?;
+                        .ok_or(EthApiError::HeaderNotFound(block_number.map(Into::into).expect("Should get latest block if block id is None")))?;
```
Comment (kpp) [inline crates/evm/src/query.rs:638]: It will panic on None

**Comment by eyusufatik** [comment]
risc0 already updated, cant update reth and alloy yet

**Pull request closed without merging**